src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -122,5 +201,41 @@ private Task<ICodingConventionContext> GetConventionContextAsync(string path, Ca
                 () => _codingConventionsManager.GetConventionContextAsync(path, cancellationToken),
                 defaultValue: EmptyCodingConventionContext.Instance);
         }
+
+        private Task OnCodingConventionsChangedAsync(object sender, CodingConventionsChangedEventArgs arg)
+        {
+            // this is a temporary workaround. once we finish the work to put editorconfig file as a part of roslyn solution snapshot,
+            // that system will automatically pick up option changes and update snapshot. and it will work regardless
+            // whether a file is opened in editor or not.
+            // 
+            // but until then, we need to explicitly touch workspace to update snapshot. and 
+            // only works for open files. it is not easy to track option changes for closed files with current model.
+            // related tracking issue - https://github.com/dotnet/roslyn/issues/26250
+            //
+            // use its own lock to remove dead lock possibility
+            ResettableDelay delay = null;
+            lock (_eventGate)
+            {
+                if (!_resettableDelay.Task.IsCompleted)
+                {
+                    _resettableDelay.Reset();
+                }
+                else
+                {
+                    // since this event gets raised for all documents that are affected by 1 editconfig file,
+                    // and since for now we make that event as whole solution changed event, we don't need to update
+                    // snapshot for each events. aggregate all events to 1.
+                    delay = new ResettableDelay(EventDelayInMillisecond);
+                    _resettableDelay = delay;
+                }
+            }
+
+            delay?.Task.ContinueWith(_ => _workspace.OnOptionChanged(),
+                CancellationToken.None,
+                TaskContinuationOptions.ExecuteSynchronously,
+                TaskScheduler.Default);
```

Comment:
sure.

Reply:
Note: this would also be much simpler if this was jsut async/await code.  I don't really understand why we are doing manual .ContinueWith coding.  That made sense in the early days of roslyn when we didn't have async/await, (or in some very low level helpers), but regular component code really shouldn't need to do this.